src/test/java/com/kodilla/ecommercee/domainTests/CartTestSuite.java

```diff
@@ -34,54 +33,53 @@ public class CartTestSuite {
     @Autowired
     private GroupRepository groupRepository;
 
-    @Autowired
-    private OrderRepository orderRepository;
-
-    User user = new User(null, "Someone", "password123", false, "blahblah", new ArrayList<>(), new ArrayList<>(), true);
+    User user = new User( "Someone", "password123","blahblah");
+    Group group = new Group("food", "things to eat");
 
     @AfterEach
     public void CleanUp() {
         cartRepository.deleteAll();
         userRepository.deleteAll();
         productRepository.deleteAll();
-        userRepository.deleteAll();
-        orderRepository.deleteAll();
+        groupRepository.deleteAll();
     }
 
     @Test
     public void testCreateCart() {
         //Given
         userRepository.save(user);
-        Cart cart = new Cart(null, user, new ArrayList<>(), true);
+        Cart cart = new Cart(user);
 
         //When
         Cart savedCart = cartRepository.save(cart);
+        Optional<Cart> retrievedCart = cartRepository.findById(savedCart.getId());
 
         //Then
-        assertNotNull(savedCart.getId());
-        assertTrue(savedCart.isActive());
+        assertNotNull(retrievedCart.get().getId());
+        assertTrue(retrievedCart.get().isActive());
     }
 
     @Test
     public void testDeleteCart() {
         //Given
         userRepository.save(user);
-        Cart cart = new Cart(null, user, new ArrayList<>(), true);
+        Cart cart = new Cart(user);
         Cart savedCart = cartRepository.save(cart);
 
         //When
         cartRepository.delete(cart);
+        Optional<Cart> retrievedCart = cartRepository.findById(savedCart.getId());
 
         //Then
-        assertNotNull(savedCart.getId());
-        assertFalse(savedCart.isActive());
+        assertNotNull(retrievedCart.get().getId());
+        assertFalse(retrievedCart.get().isActive());
     }
 
     @Test
     public void testGetCartById() {
         //Given
         userRepository.save(user);
-        Cart cart = new Cart(null, user, new ArrayList<>(), true);
+        Cart cart = new Cart(user);
         Cart savedCart = cartRepository.save(cart);
 
         //When
```

Comment:
to jest sugestia do linijek 83-90, nie wiem czemu nie chce mi się to robić. ostatnia linijka, to po prostu zamiana - pierwsza to rzecz oczekiwana, a druga to ta którą sprawdzamy czy się zgadza :)

```suggestion
         cartRepository.save(cart);

        //When
        Optional<Cart> retrievedCart = cartRepository.findById(cart.getId());

        //Then
        assertTrue(retrievedCart.isPresent());
        assertEquals(cart.getId(), retrievedCart.get().getId());
```